src/state/routing/slice.ts

```diff
@@ -23,9 +23,34 @@ const baseQuery: BaseQueryFn<GetQuoteArgs, TradeResult> = () => {
 export const routing = createApi({
   reducerPath: 'routing',
   baseQuery,
-  serializeQueryArgs: serializeGetQuoteArgs,
+  serializeQueryArgs: serializeGetQuoteQueryArgs,
   endpoints: (build) => ({
     getTradeQuote: build.query({
+      async onQueryStarted(args: GetQuoteArgs | SkipToken, { queryFulfilled }) {
+        if (args === skipToken) return
+
+        args.onQuote?.(
+          JSON.parse(serializeGetQuoteArgs(args)),
+          queryFulfilled
+            .catch((error) => {
+              const { error: queryError } = error
+              if (queryError && typeof queryError == 'object' && 'status' in queryError) {
+                const parsedError = queryError as FetchBaseQueryError
+                switch (parsedError.status) {
+                  case 'CUSTOM_ERROR':
+                  case 'FETCH_ERROR':
+                  case 'PARSING_ERROR':
```

Comment:
type/enum these?

Reply:
these are literal strings in FetchBaseQueryError, but I can instead check for `typeof parsedError.status === 'number'`

Reply:
typechecking requires that this be done without checking for `=== 'number'` - this is the most elegant way I could get past the typechecker :(